packages/opentelemetry-plugin-grpc/src/grpc.ts

```diff
@@ -0,0 +1,625 @@
+              this.moduleName,
+              this.version,
+              versionRange,
+              this.internalFileList
+            );
+          }
+          result = this._loadInternalModuleFiles(
+            this.internalFileList[versionRange],
+            basedir
+          );
+        }
+      });
+      if (Object.keys(result).length === 0) {
+        this._logger.debug(
+          'No internal file could be loaded for %s@%s',
+          this.moduleName,
+          this.version
+        );
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Load internal files from a module and set internalFilesExports
+   */
+  private _loadInternalModuleFiles(
+    extraModulesList: ModuleNameToFilePath,
+    basedir: string
+  ): ModuleExportsMapping {
+    const extraModules: ModuleExportsMapping = {};
+    if (extraModulesList) {
+      Object.keys(extraModulesList).forEach(moduleName => {
+        try {
+          this._logger.debug('loading File %s', extraModulesList[moduleName]);
+          extraModules[moduleName] = require(path.join(
+            basedir,
+            extraModulesList[moduleName]
+          ));
+        } catch (e) {
+          this._logger.error(
+            'Could not load internal file %s of module %s. Error: %s',
+            path.join(basedir, extraModulesList[moduleName]),
+            this.moduleName,
+            e.message
+          );
+        }
+      });
+    }
+    return extraModules;
+  }
+  // --- End of internal file loader stuff --- //
+
+  protected patch(): typeof grpcModule {
+    this._logger.debug(
+      'applying patch to %s@%s',
+      this.moduleName,
+      this.version
+    );
+
+    if (this._moduleExports.Server) {
+      shimmer.wrap(
+        this._moduleExports.Server.prototype,
+        'register',
+        // tslint:disable-next-line:no-any
+        this._patchServer() as any
+      );
+    }
+
+    if (this.internalFilesExports && this.internalFilesExports['client']) {
+      grpcClientModule = this.internalFilesExports['client'];
+
+      shimmer.wrap(
+        grpcClientModule,
+        'makeClientConstructor',
+        this._patchClient()
+      );
+    }
+
+    return this._moduleExports;
+  }
+  protected unpatch(): void {
+    this._logger.debug(
+      'removing patch to %s@%s',
+      this.moduleName,
+      this.version
+    );
+
+    if (this._moduleExports.Server) {
+      shimmer.unwrap(this._moduleExports.Server.prototype, 'register');
+    }
+
+    if (grpcClientModule) {
+      shimmer.unwrap(grpcClientModule, 'makeClientConstructor');
+    }
+  }
+
+  private static _toSpanStatus(status: number): Status {
+    return { code: status };
+  }
+
+  private _getSpanContext(metadata: grpcModule.Metadata): SpanContext | null {
+    const metadataValue = metadata.getMap()[GRPC_TRACE_KEY] as Buffer;
+    // Entry doesn't exist
+    if (!metadataValue) {
+      return null;
+    }
+    const spanContext = this._tracer.getBinaryFormat().fromBytes(metadataValue);
```

Comment:
I'd do `return this._tracer.getBinaryFormat().fromBytes(metadataValue);`, below check looks superfluous.